File: src/main/java/no/nav/domain/pensjon/regler/beregning2011/UforetrygdOrdiner.java
package no.nav.domain.pensjon.regler.beregning2011;

import no.nav.domain.pensjon.regler.beregning.Ytelseskomponent;
import no.nav.domain.pensjon.regler.kode.YtelsekomponentTypeCti;

/**
 * Ytelseskomponent for uføretrygd.
 * 
 * @author Steinar Hjellvik (Decisive) PK-6172
 * @author Aasmund Nordstoga (Accenture) PK-9029
 * @author Tatyana Lochehina PK-13673
 */
public class UforetrygdOrdiner extends Ytelseskomponent {


    /**
     * Brukers minsteytelse.
     */
    private Minsteytelse minsteytelse;

    /**
     * Brukers uføretrygd før inntektsavkorting.
     */
    private EgenopptjentUforetrygd egenopptjentUforetrygd;

    /**
     * Angir om egenopptjentUforetrygd er best.
     */
    private boolean egenopptjentUforetrygdBest;

    /**
     * Angir detaljer rundt inntektsavkortingen.
     */
    private AvkortingsinformasjonUT avkortingsinformasjon;

    /**
     * Akkumulert netto hittil i året eksklusiv måned for beregningsperiodens fomDato.
     */
    private int nettoAkk;

    /**
     * gjenstående beløp brukeren har rett på for året som beregningsperioden starter,
     * og inkluderer måneden det beregnes fra.
     */
    private int nettoRestAr;

    /**
     * Inntektsavkortningsbeløp per år, før justering med differansebeløp
     */
    private int avkortningsbelopPerAr;

    /**
     * Utrykker avviket mellom lignet og forventet beløp ved etteroppgjør.
     */
    private double periodisertAvvikEtteroppgjor;

    /**
     * Angir fradragPerAr dersom det ikke hadde vært arbeidsforsøk i perioden.
     */
    private double fradragPerArUtenArbeidsforsok;

    /**
     * Årsbeløp for delytelsen fra tidligere vedtak (fra tilsvarende beregningsperiode)
     */
    private int tidligereBelopAr;

    public UforetrygdOrdiner() {
        super();
        ytelsekomponentType = new YtelsekomponentTypeCti("UT_ORDINER");
    }

}

File: src/main/kotlin/no/nav/pensjon/regler/domain/beregning2011/UforetrygdOrdiner.kt
package no.nav.pensjon.regler.domain.beregning2011

import no.nav.pensjon.regler.domain.beregning.Ytelseskomponent
import no.nav.pensjon.regler.domain.kode.YtelsekomponentTypeCti

/**
 * Ytelseskomponent for uføretrygd.
 */
class UforetrygdOrdiner : Ytelseskomponent(), UforetrygdYtelseskomponent {
    /**
     * Brukers minsteytelse.
     */
    var minsteytelse: Minsteytelse? = null

    /**
     * Brukers uføretrygd før inntektsavkorting.
     */
    var egenopptjentUforetrygd: EgenopptjentUforetrygd? = null

    /**
     * Angir om egenopptjentUforetrygd er best.
     */
    var egenopptjentUforetrygdBest = false

    /**
     * Angir detaljer rundt inntektsavkortingen.
     */
    var avkortingsinformasjon: AvkortingsinformasjonUT? = null

    /**
     * Akkumulert netto hittil i året eksklusiv måned for beregningsperiodens fomDato.
     */
    var nettoAkk = 0

    /**
     * gjenstående beløp brukeren har rett på for året som beregningsperioden starter,
     * og inkluderer måneden det beregnes fra.
     */
    var nettoRestAr = 0

    /**
     * Inntektsavkortningsbeløp per år, før justering med differansebeløp
     */
    var avkortningsbelopPerAr = 0

    /**
     * Utrykker avviket mellom lignet og forventet beløp ved etteroppgjør.
     */
    var periodisertAvvikEtteroppgjor = 0.0

    /**
     * Angir fradragPerAr dersom det ikke hadde vært arbeidsforsøk i perioden.
     */
    var fradragPerArUtenArbeidsforsok = 0.0

    /**
     * årsbeløp for delytelsen fra tidligere vedtak (fra tilsvarende beregningsperiode)
     */
    override var tidligereBelopAr = 0

    init {
        ytelsekomponentType = YtelsekomponentTypeCti("UT_ORDINER")
    }
}